src/librbd/api/Image.cc

```diff
@@ -760,10 +762,19 @@ int Image<I>::remove(IoCtx& io_ctx, const std::string &image_name,
     } else {
       // attempt to pre-validate the removal before moving to trash and
       // removing
-      r = pre_remove_image<I>(io_ctx, image_id);
+      bool attached_children = false;
+      if (config.get_val<bool>("rbd_move_parent_to_trash_on_remove")) {
+        r = pre_remove_image<I>(io_ctx, image_id, &attached_children);
+      } else {
+        r = pre_remove_image<I>(io_ctx, image_id, nullptr);
+      }
       if (r < 0 && r != -ENOENT) {
         return r;
       }
+      if (attached_children) {
+        // keep the image in the trash until the last child is removed
+        trash_image_source = RBD_TRASH_IMAGE_SOURCE_USER;
```

Comment:
I agree that it would be nice to have a new ```_USER_PARENT``` (or similar) enum. The ```trash purge``` could attempt  a best effort to remove parents but not fail if it cannot.